src/every.ts

```diff
@@ -0,0 +1,100 @@
+import type {
+  Pred,
+  PredIndexed,
+  PredIndexedOptional,
+  TypePred,
+  TypePredIndexed,
+} from "./_types";
+import { purry } from "./purry";
+
+/**
+ * Checks if every element of `data` passed the predicate `predicate`.
+ *
+ * @param data - The Array to check.
+ * @param predicate - A predicate function to run.
+ * @returns A `boolean` indicating if every element in `data` passes the predicate `pred`.
+ * @signature
+ *    R.every(arr, predicate)
+ * @example
+ *    R.every([1, 2, 3], R.isNumber) // => true
+ *    R.every(['a', 0, 'b'], R.isString) // => false
+ * @dataFirst
+ * @indexed
+ * @pipeable
+ * @category Array
+ */
+export function every<T, S extends T>(
+  data: ReadonlyArray<T>,
+  predicate: TypePred<T, S>,
+): data is Array<S>;
+export function every<T>(
+  data: ReadonlyArray<T>,
+  predicate: Pred<T, boolean>,
+): boolean;
+
+/**
+ * Checks if every element of `data` passed the predicate `predicate`.
+ *
+ * NOTE: If you want to use this inside `conditional` predicates and preserve proper type inference,
+ * make sure to define the final type guard outside of the pipe. This is due to Typescript's limitation
+ * in generic type inference. See examples below for details.
+ *
```

Comment:
**question:** what do you feel about dropping this? i think this is a problem with `conditional` and not `every`, and so notes like this should go into usage of `conditional` (because they also apply to other predicates)

Reply:
also, i would say things like this should go in one of the conditional tests as like, a known issue

Reply:
(rather than the jsdoc, which appears every time we hover over the function)

Reply:
Removed 👍